src/constructors/generalized-petersen.c

```diff
@@ -0,0 +1,45 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "igraph_constructors.h"
+
+#include "igraph_interface.h"
+
+
+int igraph_generalized_petersen(igraph_t *igraph, igraph_integer_t n, igraph_integer_t k) {
```

Comment:
Looks good so far! Two minor things are to change the return type to `igraph_error_t` (this is new on the `develop` branch) and to name the argument `graph` instead of `igraph`. Let us know if you have any questions as you go.

Reply:
Got it, thanks! I've fixed these on the latest commit.